client/src/lib/paymentCalc.ts

```diff
@@ -1,58 +1,104 @@
 import { ClientConstants } from "../constants";
-import { AbiType, AxiomV2ClientOverrides, AxiomV2QueryOptions } from "../types";
+import { AbiType, AxiomV2ClientOverrides, AxiomV2ClientOptions } from "../types";
 import { PublicClient } from "viem";
-import { getAxiomV2QueryAddress } from "./address";
-import { getAxiomV2Abi } from "./abi";
+import { getAxiomV2QueryAddress, getOpStackGasPriceOracleAddress, getOpStackL1BlockAttributesAddress } from "./address";
+import { getAxiomV2Abi, getOpStackGasPriceOracleAbi, getOpStackL1BlockAttributesAbi } from "./abi";
+import { isArbitrumChain, isMainnetChain, isOpStackChain, isScrollChain } from "./chain";
+import { publicActionsL2 } from 'viem/op-stack';
 
 export async function calculatePayment(
-  axiomV2QueryAddr: string,
+  chainId: string,
   publicClient: PublicClient,
-  options?: AxiomV2QueryOptions
+  options?: AxiomV2ClientOptions,
 ): Promise<bigint> {
+  const axiomV2QueryAddr = options?.overrides?.queryAddress ?? getAxiomV2QueryAddress(chainId);
+
+  // Convert callback gas limit to wei
+  const callbackGasLimit = BigInt(options?.callbackGasLimit ?? ClientConstants.DEFAULT_CALLBACK_GAS_LIMIT);
```

Comment:
depending on how the calculatePayment function is called, the `options` argument can be undefined. when `calculatePayment` called inside the client SDK, options is filled out the value passed into it, but if it's called externally, the caller may not have `options` set

Reply:
and yes, not doing any conversion just handling when values are undefined

Reply:
When would it be called externally?

Reply:
currently in tests, but it's exported by the client package for devs to call if they want to pass in their own values for their own logic. we can definitely make options non-optional and have devs to pass in values instead of using defaults if nothing is passed in

Reply:
Yeah, I think that's safer (and we can have default values exported as constants).